src/main/java/de/caritas/cob/agencyservice/api/service/TenantHibernateInterceptor.java

```diff
@@ -13,16 +15,52 @@
 @Slf4j
 public class TenantHibernateInterceptor extends EmptyInterceptor {
 
+  private static Boolean multiTenancyEnabled;
+
   @Override
   public void preFlush(Iterator entities) {
-    Object entity;
-    while (entities.hasNext()) {
-      entity = entities.next();
-      if (entity instanceof TenantAware) {
-        ((TenantAware) entity).setTenantId(TenantContext.getCurrentTenant());
+
+    if (multiTenancyEnabled == null) {
+      multiTenancyEnabled = resolveMultiTenancy();
+    }
+
+    if (multiTenancyEnabled) {
```

Comment:
I would prefer a call to MultitenancyPropertyResolver.isMultitenancyEnabled(), which would then have the value cached as static field

Reply:
explained my ideas in past comment.